BookingQueueSubscriber/BookingQueueSubscriber.Services/MessageHandlers/EndpointUpdatedHandler.cs

```diff
@@ -11,26 +13,60 @@ namespace BookingQueueSubscriber.Services.MessageHandlers
     public class EndpointUpdatedHandler : IMessageHandler<EndpointUpdatedIntegrationEvent>
     {
         private readonly IVideoApiService _videoApiService;
+        private readonly ILogger<EndpointUpdatedHandler> _logger;
+        private const int RetryLimit = 3;
+        private const int RetrySleep = 3000;
 
-        public EndpointUpdatedHandler(IVideoApiService videoApiService)
+        public EndpointUpdatedHandler(IVideoApiService videoApiService, ILogger<EndpointUpdatedHandler> logger)
         {
             _videoApiService = videoApiService;
+            _logger = logger;
         }
 
         public async Task HandleAsync(EndpointUpdatedIntegrationEvent eventMessage)
         {
-            var conference = await _videoApiService.GetConferenceByHearingRefId(eventMessage.HearingId);
             ParticipantDetailsResponse defenceAdvocate = null;
+            ConferenceDetailsResponse conference = null;
+
             if (!string.IsNullOrEmpty(eventMessage.DefenceAdvocate))
             {
-                defenceAdvocate = conference.Participants.Single(x => x.ContactEmail ==
-                    eventMessage.DefenceAdvocate);
+                for (var retry = 0; retry <= RetryLimit; retry++)
+                {
+                    conference = await _videoApiService.GetConferenceByHearingRefId(eventMessage.HearingId);
+
+                    if (conference == null)
+                    {
+                        _logger.LogError("Unable to find conference by hearing id {HearingId}", eventMessage.HearingId);
```

Comment:
we probably also need a `break;` after logging the error to exit the retry loop if the conference doesn't exist

Reply:
yes I think so, as per the current behaviour. This is for updating endpoints - ie for conferences that already exist, so if the conference wasn't found then I don't think there's a need to keep looping.